rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMEvent.java

```diff
@@ -136,4 +145,27 @@ public String toString() {
     public Object clone() throws CloneNotSupportedException {
         return super.clone();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+
+        RMEvent rmEvent = (RMEvent) o;
+
+        return counter == rmEvent.counter;
+    }
+
+    @Override
+    public int hashCode() {
+        return (int) (counter ^ (counter >>> 32));
+    }
+
+    @Override
+    public int compareTo(RMEvent event) {
+        return (int) this.getCounter() - (int) event.getCounter();
```

Comment:
I think that casting the two counters before computing the difference is a mistake. We could end up having to truncate the long and computing a false difference, could we? Maybe it would be safer to do `return (int) (this.getCounter() - event.getCounter());`.

Also, we can use field access instead of getters for `counter`.

Reply:
You are right, i will do this